### DISCUSSION TITLE
Pyright flaging mixed keys dictionary as an error

### DISCUSSION THREAD

**Discussion by Pilgrim1379** [microsoft/pyright#9187] [Q&A] [answered]
The following is a copy paste from [FastAPI - Query Parameters and String Validations](https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#__tabbed_1_1l)

```python
from typing import Annotated
from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/items/")
async def read_items(q: Annotated[str | None, Query(max_length=50)] = None):
    results = {"items": [{"items_id": "Foo"}, {"item_id": "Bar"}]}
    if q:
        results.update({"q": q})
    return results
```
`Pyright` complains like below:
```
1. No overload for "update" match the provided arguments
2. Argument of type "dict[str, str]" cannot be assigned to parameter "m" of type "Iterable[tuple[str, list[dict[str, str]]]]" in function "update"
"Literal['q']" is not assignable to "tuple[str, list[dict[str, str]]]"
```
It seems `pyright` is overly strict, expecting that all values in `results` should match the structure of the `items` key (i.e., a list of dictionaries), when in fact you can update the dictionary with any new key-value pair (e.g., a string).

How can I configure `pyright` to not flag such line as an error?

**Comment by erictraut** [comment, marked as answer]
This is a legitimate type violation, so pyright is correct to generate an error here. You can suppress type errors with a `# type: ignore` or `# pyright: ignore` comments, but it's better to fix your code to eliminate the type violation. Otherwise you defeat the purpose of a type checker.

If your intent is for `results` to be a `dict` that accepts values of any type, you can declare its type as `dict[str, object]` by providing a type annotation.

```python
results: dict[str, object] = ...
```

For more information about type declarations and type inference, refer to the [pyright documentation](https://microsoft.github.io/pyright/#/type-inference).
